packages/opentelemetry-node/test/registration.test.ts

```diff
@@ -21,11 +21,19 @@ import {
   trace,
   ProxyTracerProvider,
 } from '@opentelemetry/api';
-import { AsyncHooksContextManager } from '@opentelemetry/context-async-hooks';
+import {
+  AsyncHooksContextManager,
+  AsyncLocalStorageContextManager,
+} from '@opentelemetry/context-async-hooks';
 import { NoopContextManager } from '@opentelemetry/context-base';
 import { CompositePropagator } from '@opentelemetry/core';
 import * as assert from 'assert';
 import { NodeTracerProvider } from '../src';
+import * as semver from 'semver';
+
+const DefaultContextManager = semver.gte(process.version, '14.8.0')
```

Comment:
should we rather check if AsyncLocalStorageContextManager is available (feature detection) instead of checking the version using semver. If you already require that it means it will be undefined for lower versions 
```
const DefaultContextManager = AsyncLocalStorageContextManager || AsyncHooksContextManager
```
should be enough ?

Reply:
In the description he mentioned that there is a bug in versions before 14.8. Feature detection would still detect that local storage is available, but he suggests we would want to use the async hooks version anyways.

Reply:
sry haven't spotted that